cli/azd/pkg/tools/azcli/webapp.go

```diff
@@ -75,23 +76,117 @@ func (cli *azCli) appServiceRepositoryHost(
 	return hostName, nil
 }
 
+func checkWebAppDeploymentStatus(
+	res armappservice.WebAppsClientGetProductionSiteDeploymentStatusResponse,
+) (string, error) {
+	properties := res.CsmDeploymentStatus.Properties
+	inProgressNumber := int(*properties.NumberOfInstancesInProgress)
+	successNumber := int(*properties.NumberOfInstancesSuccessful)
+	failNumber := int(*properties.NumberOfInstancesFailed)
+	totalNumber := inProgressNumber + successNumber + failNumber
+	failLogs := properties.FailedInstancesLogs
+	errorString := ""
+
+	switch *properties.Status {
+	case armappservice.DeploymentBuildStatusRuntimeSuccessful:
+		return "", nil
+	case armappservice.DeploymentBuildStatusRuntimeFailed:
+		if successNumber > 0 {
+			errorString += fmt.Sprintf("Site started with errors: %d/%d instances failed to start successfully\n",
+				failNumber, totalNumber)
+		} else if totalNumber > 0 {
+			errorString += fmt.Sprintf("Deployment failed because the runtime process failed. In progress instances: %d, "+
+				"Successful instances: %d, Failed Instances: %d\n",
+				inProgressNumber, successNumber, failNumber)
+		}
+
+		errors := properties.Errors
+
+		for _, err := range errors {
+			if err.Message != nil {
+				errorString += fmt.Sprintf("Error: %s\n", *err.Message)
+			}
+		}
+
+		for _, log := range failLogs {
+			errorString += fmt.Sprintf("Please check the runtime logs for more info: %s\n", *log)
+		}
+
+		return "", fmt.Errorf(errorString)
+	case armappservice.DeploymentBuildStatusBuildFailed:
+		errorString += "Deployment failed because the build process failed\n"
+		errors := properties.Errors
+
+		for _, err := range errors {
+			if err.Message != nil {
+				errorString += fmt.Sprintf("Error: %s\n", *err.Message)
+			}
+		}
+
+		for _, log := range failLogs {
+			errorString += fmt.Sprintf("Please check the build logs for more info: %s\n", *log)
+		}
+
+		return "", fmt.Errorf(errorString)
+	// Default case for the rest statuses, they shouldn't appear as a final response
+	default:
+		errorString += fmt.Sprintf("Deployment failed with status: %s\n", *properties.Status)
+		errors := properties.Errors
+
+		for _, err := range errors {
+			if err.Message != nil {
+				errorString += fmt.Sprintf("Error: %s\n", *err.Message)
+			}
+		}
+
+		for _, log := range failLogs {
+			errorString += fmt.Sprintf("Please check the logs for more info: %s\n", *log)
+		}
+
+		return "", fmt.Errorf(errorString)
```

Comment:
Looks like this code is repeated in every case of the switch?  
Maybe move it outside to after the switch?